test/test_typed_array_timestamp.cpp

```diff
@@ -0,0 +1,237 @@
+// Copyright 2024 Man Group Operations Limited
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or mplied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <chrono>
+#include <compare>
+#include <cstddef>
+#include <cstdint>
+#include <iostream>  // For Doctest
+#include <sstream>
+#include <string>
+#include <type_traits>
+
+#if defined(SPARROW_USE_DATE_POLYFILL)
+#include <date/tz.h>
+#else
+namespace date = std::chrono; 
+#endif
+
+#include "sparrow/typed_array.hpp"
+#include "sparrow/data_type.hpp"
+
+#include "array_data_creation.hpp"
+#include "doctest/doctest.h"
+
+namespace
+{
+    constexpr std::size_t test_n = 10;
+    constexpr std::size_t test_offset = 1;
+    const std::vector<size_t> false_bitmap = {9};
+    using sys_time = std::chrono::time_point<std::chrono::system_clock, std::chrono::nanoseconds>;
+}
+
+TEST_SUITE("typed_array_timestamp")
+{
+    TEST_CASE("constructor with parameter")
+    {
+        const auto array_data = sparrow::test::make_test_array_data<sparrow::timestamp>(test_n, test_offset);
+        const sparrow::typed_array<sparrow::timestamp> ta{array_data};
+        CHECK_EQ(ta.size(), test_n - test_offset);
+    }
+
+    // Element access
+
+    TEST_CASE("at")
+    {
+        const auto array_data = sparrow::test::make_test_array_data<sparrow::timestamp>(test_n, test_offset, false_bitmap);
+        sparrow::typed_array<sparrow::timestamp> ta{array_data};
+        for (typename sparrow::typed_array<sparrow::timestamp>::size_type i = 0; i < ta.size() - 1; ++i)
+        {
+            CHECK_EQ(ta.at(i).value(), sparrow::timestamp(date::sys_days(date::year(1970)/date::January/date::day(1)) + date::days(i + 1)));
```

Comment:
Can you extract `date::sys_days(date::year(1970)/date::January/date::day(1))` in a constant to ease the readibility ?